### PR TITLE
Repackage as /v2 and change builtin actor code CIDs to fil/2

### LINKED ISSUE #1105: Re-package as specs-actors/v2
In order to support versioning actors in the testnet, the new behaviour needs to be separately importable from the code in the v0.9 branch.

### LINKED ISSUE #1107: Change actor code CIDs to reflect version 2
Change the actor code CIDs to replace the version `1` with `2` (the code on the v0.9 branch will retain the `1`).

The future migration function will need to update them all in-place in the state tree.

### PR DESCRIPTION
**Pull request by anorth**
This PR changes the package path for the repo to github.com/filecoin-project/specs-actors/v2. This is in preparation for eventually cutting the mainnet-candidate release-2.0 branch and release series. We intend to version functional upgrades to actors with release branches and corresponding Go package major versions, such that a downstream project can import multiple different versions at once.

This repackaging does not imply that no more functional changes should be landing; we still have some fixes to land before mainnet and consumers should not assume state or functionality is fixed until at least the release branch is cut.

This is a purely mechanical renaming of the package. I would have used `v1` (#1088) but go mod doesn't differentiate that from version zero and wouldn't allow importing both at once 😦 .

Note: in order to ease testing an upgrade between the 0.9 branch and this, a subsequent change will import specs-actors v0.9.3 into this branch (master) and then alias a bunch of types that haven't changed onto their 0.9 versions.

- [x] Wait for #1109 then rebase on master

Closes #1105. Closes #1107. FYI @whyrusleeping @magik6k @arajasek @raulk @hannahhoward

### CODE REVIEW

File: actors/builtin/market/deal.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type DealProposal struct {
 	Provider     addr.Address
 
 	// Label is an arbitrary client chosen label to apply to the deal
-	// TODO: Limit the size of this: https://github.com/filecoin-project/specs-actors/issues/897
+	// TODO: Limit the size of this: https://github.com/filecoin-project/specs-actors/v2/issues/897
```
Comment (ZenGround0) [inline actors/builtin/market/deal.go:37]: overzealous sed

File: actors/builtin/miner/deadline_assignment.go [hunk 2]
```diff
@@ -157,7 +157,7 @@ func (dah *deadlineAssignmentHeap) Less(i, j int) bool {
 
 	// Finally, fallback on the deadline index.
 	// TODO: Randomize by index instead of simply sorting.
-	// https://github.com/filecoin-project/specs-actors/issues/432
+	// https://github.com/filecoin-project/specs-actors/v2/issues/432
```
Comment (ZenGround0) [inline actors/builtin/miner/deadline_assignment.go:160]: overzealous sed

File: actors/builtin/network.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 // TODO: In lieu of a real configuration mechanism for this value, we'd like to make it a var so that implementations
 // can override it at runtime. Doing so requires changing all the static references to it in this repo to go through
 // late-binding function calls, or they'll see the "wrong" value.
-// https://github.com/filecoin-project/specs-actors/issues/353
+// https://github.com/filecoin-project/specs-actors/v2/issues/353
```
Comment (ZenGround0) [inline actors/builtin/network.go:16]: nit

File: actors/util/adt/store.go [hunk 4]
```diff
@@ -57,7 +57,7 @@ func (r rtStore) Context() context.Context {
 
 func (r rtStore) Get(_ context.Context, c cid.Cid, out interface{}) error {
 	// The Go context is (un/fortunately?) dropped here.
-	// See https://github.com/filecoin-project/specs-actors/issues/140
+	// See https://github.com/filecoin-project/specs-actors/v2/issues/140
```
Comment (ZenGround0) [inline actors/util/adt/store.go:60]: nit

File: actors/util/adt/store.go [hunk 5]
```diff
@@ -66,7 +66,7 @@ func (r rtStore) Get(_ context.Context, c cid.Cid, out interface{}) error {
 
 func (r rtStore) Put(_ context.Context, v interface{}) (cid.Cid, error) {
 	// The Go context is (un/fortunately?) dropped here.
-	// See https://github.com/filecoin-project/specs-actors/issues/140
+	// See https://github.com/filecoin-project/specs-actors/v2/issues/140
```
Comment (ZenGround0) [inline actors/util/adt/store.go:69]: nit

**Review by ZenGround0 — APPROVED** [review]

**Comment by anorth** [comment]
Thanks, I found and fixed more comments caught up by mistake

**Pull request merged**